app/src/main/java/com/infomaniak/mail/ui/bottomSheetDialogs/ScheduleSendBottomSheetDialog.kt

```diff
@@ -0,0 +1,224 @@
+            val draftResource = navigationArgs.draftResource
+            val lastSelectedScheduleDate = localSettings.lastSelectedScheduleDate
+
+            if (navigationArgs.isAlreadyScheduled) {
+                if (draftResource != null && lastSelectedScheduleDate != null) {
+                    trackScheduleSendEvent("lastSchedule")
+                    mainViewModel.rescheduleDraft(draftResource, Date(lastSelectedScheduleDate))
+                }
+            } else {
+                lastSelectedScheduleDate?.let {
+                    trackScheduleSendEvent("lastSchedule")
+                    newMessageViewModel.setScheduleDate(Date(it))
+                    newMessageViewModel.triggerSendMessage()
+                }
+            }
+        }
+
+        if (newMessageViewModel.currentMailbox.isFree) {
+            customScheduleItem.setOnClickListener {
+                safeNavigate(
+                    resId = R.id.upgradeProductBottomSheetDialog,
+                    currentClassName = ScheduleSendBottomSheetDialog::class.java.name,
+                )
+            }
+        } else {
+            customScheduleItem.setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    mainViewModel.showSelectDateAndTimeForScheduleDialog()
+                } else {
+                    newMessageViewModel.showSelectDateAndTimeForScheduleDialog()
+                }
+            }
+        }
+
+        fun createScheduleItem(schedule: Schedule): ActionItemView = ActionItemView(this.context).apply {
+            setTitle(schedule.scheduleTitleRes)
+            setDescription(mostDetailedDate(context, date = schedule.date, format = FORMAT_DATE_DAY_MONTH))
+            setIconResource(schedule.scheduleIconRes)
+            setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    navigationArgs.draftResource?.let {
+                        trackScheduleSendEvent(schedule.matomoValue)
+                        mainViewModel.rescheduleDraft(draftResource = it, schedule.date)
+                    }
+                } else {
+                    trackScheduleSendEvent(schedule.matomoValue)
+                    newMessageViewModel.setScheduleDate(schedule.date)
+                    newMessageViewModel.triggerSendMessage()
+                }
+            }
+        }
+
+        val timeToDisplay = TimeToDisplay.getTimeToDisplayFromDate()
+        Schedule.entries.filter { schedule -> timeToDisplay in schedule.timeToDisplay }.forEach { schedule ->
+            scheduleItems.addView(createScheduleItem(schedule))
+        }
+
+        val shouldDisplayDivider = lastScheduleItem.isVisible
+        (scheduleItems.children.first() as ActionItemView).setDividerVisibility(shouldDisplayDivider)
+    }
+
+    companion object {
+        const val MIN_SCHEDULE_DELAY_MINUTES = 5
+        const val MAX_SCHEDULE_DELAY_YEARS = 10
+    }
```

Comment:
These constants are more often than not used in another file than this one. Would it not be better to write them where they'll mainly be used?